### ISSUE TITLE
Location of 'Convert session to NSM file format'

### ISSUE THREAD

**Issue by Stazed** [Houston4444/RaySession#232]
I cannot seem to find the converted NSM session when trying: File/Utilities/Convert session to NSM file format.
I wanted to try the other session managers for testing... I would guess it should be in the ~/.local/share/nsm directory, but not there or any where else I could find.
Thanks.

**Comment by Houston4444** [comment]
Fast answer (I will check tonight if it is correct), IIRC , session is converted on place, you have to move the session folder to `~/.local/share/NSM Sessions` to use it in new-session-manager or Agordejo.

**Comment by Stazed** [comment]
Copying to ~/.local/share/nsm does not seem to work. There is no session.nsm file created. Also if I check the 'Use NSM jackpatch' box, there is no JACKPatch.xxxxx.jackpatch file created.

**Comment by Houston4444** [comment]
I can't reproduce this bug.
Could you try to do the same in a terminal and report the output ? The script is here: https://github.com/Houston4444/RaySession/blob/master/utility-scripts/session_ray_to_nsm.sh (in the source code, in utility-scripts). 
You just have to launch it with your session name (or path relative to your sessions root) as argument.

**Comment by Stazed** [comment]
This is what I get..
```
/home/sspresto/Desktop/session_ray_to_nsm.sh Consequences
Option bookmark_session_folder is not currently used
No session to close.
-- Opening session 'Consequences' --
waiting announce from 4 clients...
  Non-Mixer-XT (m_xt): launched
  Non-Mixer-XT (m_xt): announced
  seq42 (seq42): launched
  seq42 (seq42): announced
  seq42 (seq42): project loaded
  Non-Timeline-XT (t_xt_2): launched
  JACK Connections (patch): launched
  Non-Timeline-XT (t_xt_2): announced
  JACK Connections (patch): announced
waiting for 3 clients to load their project...
  JACK Connections (patch): project loaded
  Non-Timeline-XT (t_xt_2): project loaded
  Non-Mixer-XT (m_xt): project loaded
session 'Consequences' is loaded.
No session is loaded, impossible to take snapshot
No session to list clients !
No session to close.
Done.
```

**Comment by Stazed** [comment]
Hey, just tried this on my Ubuntu Studio distro and it works. So it seems to be specific to my Manjaro distro. I manually converted the raysessions that I wanted to test, so this issue is low priority to me and I can always convert them on UB Studio. Not sure if it is worth the time to dig into at this point.

**Comment by Houston4444** [comment]
mmhh, your problem highlights another one, so, to me, it is quite annoying.
Indeed, your terminal says:

```
session 'Consequences' is loaded.
No session is loaded, impossible to take snapshot
```

It is not normal, and perhaps other scripts are not working well.

Could you please try on Manjaro, theses 3 command lines and post here the result.

```
ray_control open_session_off any_session
cat /tmp/RaySession/multi-daemon.xml
ray_control take_snapshot anything_snapshot
```

Thanks.

**Comment by Stazed** [comment]
```
$ ray_control open_session_off any_session
--- Daemon started at port 16187 ---
-- Opening session 'any_session' --
session 'any_session' is loaded.
```
```
$ cat /tmp/RaySession/multi-daemon.xml
<Daemons>
 <Daemon net_daemon_id="681291483" version="0.14.3" session_path="/home/sspresto/Ray Sessions/any_session" pid="1603" user="sspresto" local_gui_pids="" has_gui="0" not_default="0" root="/home/sspresto/Ray Sessions" port="16187"/>
</Daemons>
```
```
$ ray_control take_snapshot anything_snapshot
snapshot started...
git exit with an error code.
tag -a 2024_8_9_19_18_51 -m ray
git exit with an error code.
tag -a 2024_8_9_19_18_51 -m ray
```

**Comment by Stazed** [comment]
I just tried the conversion on my other computer with Manjaro and it worked! The only difference I can think of that might be causing the problem is the other Manjaro uses KDE Plasma desktop, and the one that does not work XFCE.
Ubuntu Studio also has KDE Plasma.

I suspect that the Manjaro that does not work might be the result of some install remove of KDE that I did a while back or some other changes I made as I use it for all kinds of testing. Probably specific to the computer...

**Comment by Houston4444** [comment]
Ok thanks. The last error is not the same and I can reproduce it, it is simply the fact that 'take_snapshot' fails if the session git project has not been initialized (it simply should init it if it is not the case of course).

Testing this, I found a very important bug that I have just fixed, 'open_session_off' was opening session normally, which make this script fails (and not only this one). TBH, seeing the code, I don't understand how it could have work, it is probably due to a python change.

I don't know how your problem happens, but of course, RaySession should works the same on Xfce !
